docs/resources/mapreduce_cluster.md

```diff
@@ -485,6 +489,27 @@ The `nodes` block supports:
   -> `DBService` is a basic component of a cluster. Components such as Hive, Hue, Oozie, Loader, and Redis, and Loader
    store their metadata in DBService, and provide the metadata backup and restoration functions by using DBService.
 
+<a name="component_configurations"></a>
+The `component_configs` block supports:
+
+* `name` - (Optional, String, ForceNew) Specifies the component name of the cluster which has installed.
+  Changing this will create a new MapReduce cluster resource.
+
+* `configs` - (Optional, List, ForceNew) Specifies the configuration of component installed.
+  The [object](#component_configuration) structure is documented below.
+
+<a name="component_configuration"></a>
+The `configs` block supports:
+
+* `key` - (Optional, String, ForceNew) Specifies the configuration item key of component installed.
+  Changing this will create a new MapReduce cluster resource.
+
+* `value` - (Optional, String, ForceNew) Specifies the configuration item value of component installed.
+  Changing this will create a new MapReduce cluster resource.
+
+* `config_file_name` - (Optional, String, ForceNew) Specifies the configuration file name of component installed.
+  Changing this will create a new MapReduce cluster resource.
```

Comment:
these parameters are required in the code

Reply:
fixed